docs/types.rst

```diff
@@ -0,0 +1,145 @@
+.. index:: type
+
+.. _types:
+
+*****
+Types
+*****
+
+Viper is a statically typed language, which means that the type of each
+variable (state and local) needs to be specified or at least known at
+compile-time. Viper provides several elementary types which can be combined
+to form complex types.
+
+In addition, types can interact with each other in expressions containing
+operators.
+
+Value Types
+===========
+
+The following types are also called value types because variables of these
+types will always be passed by value, i.e. they are always copied when they
+are used as function arguments or in assignments.
+
+.. index:: ! bool, ! true, ! false
+
+Booleans
+--------
+
+``bool``: The possible values are constants ``true`` and ``false``.
+
+Operators:
+
+*  ``not`` (logical negation)
+*  ``and`` (logical conjunction, "&&")
+*  ``or`` (logical disjunction, "||")
+*  ``==`` (equality)
+*  ``not ... == ... `` (inequality)
+*  ``!=`` (inequality)
+
+The operators ``or`` and ``and`` apply the common short-circuiting rules. This means that in the expression ``f(x) or g(y)``, if ``f(x)`` evaluates to ``true``, ``g(y)`` will not be evaluated even if it may have side-effects.
+
+.. index:: ! uint, ! int, ! integer
+
+
+Integers
+--------
+
+``num``:  a signed integer strictly between -2\*\*128 and 2\*\*128.
+
+Operators:
+
+* Comparisons: ``<=``, ``<``, ``==``, ``!=``, ``>=``, ``>`` (evaluate to ``bool``)
+* Arithmetic operators: ``+``, ``-``, unary ``-``, unary ``+``, ``*``, ``/``, ``%`` (remainder)
+
+
+Decimals
+--------
+``decimal``:  a decimal fixed point value with the integer component being a signed integer strictly between -2\*\*128 and 2\*\*128 and the fractional component being ten decimal places
+
+
+Time
+-----
+``timestamp``:  a timestamp value
+
+``timedelta``:  a number of seconds (note: two timedeltas can be added together, as can a timedelta and a timestamp, but not two timestamps)
```

Comment:
What underlying data types are `timedelta` and `timestamp`? (i.e. max `2**127` seconds, or whatever)